MaxText/standalone_checkpointer.py

```diff
@@ -93,22 +93,57 @@ def checkpoint_loop(config, state=None):
         ckpt_write_time.append([jax.process_index(), checkpoint_write_time])
         if jax.process_index() == 0:
           max_logging.log(f"STANDALONE CHECKPOINTER : Checkpoint saved in {end_time - start_time} ,step {step}, on host 0")
+    if jax.process_index() == 0:
+      elapsed_time = datetime.datetime.now() - start_time
+      time_to_wait = config.step_time_seconds - elapsed_time.total_seconds()
+      if time_to_wait > 0:
+        max_logging.log(f"Waiting {time_to_wait} seconds to reach step time of {config.step_time_seconds} seconds for step {step}")
+        time.sleep(time_to_wait)
+    jax.experimental.multihost_utils.sync_global_devices("Barrier after step")
           
   if config.gcs_metrics_bucket:
-    max_logging.log(f"Uploading metrics to GCS bucket {config.gcs_metrics_bucket} on host {jax.process_index()}")
+    max_logging.log(f"Uploading write metrics to GCS bucket {config.gcs_metrics_bucket} on host {jax.process_index()}")
     
     base_name = f"{jax.process_index()}.csv"
     storage_utils.upload_csv(
         config.gcs_metrics_bucket, 
-        os.path.join(config.run_name, CHECKPOINT_RESTORE_TIME_DIRECTORY, base_name), 
-        ckpt_read_time
+        os.path.join(config.run_name, CHECKPOINT_WRITE_TIME_DIRECTORY, base_name), 
+        ckpt_write_time
     )
+    max_logging.log(f"Finished uploading write metrics to GCS bucket {config.gcs_metrics_bucket} on host {jax.process_index()}  for run {config.run_name}")
+
+  for step in np.arange(start_step, config.steps):
+    if checkpoint_manager is not None:
+      start_time = datetime.datetime.now()
```

Comment:
can we move this tighter after the "barrier before restore"?

Reply:
Done